### PR TITLE
[bitnami/mxnet] Standardisation + Refactor

### PR DESCRIPTION
**Pull request by migruiz4**
**Description of the change**

This PR standardizes the bitnami/mxnet chart to be in line with the rest of the catalog.

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by migruiz4** [comment]
Pending: Update README.md with upgrade nodes

**Review by marcosbc — DISMISSED** [review]

File: bitnami/mxnet/templates/_helpers.tpl [hunk 1]
```diff
@@ -48,21 +48,43 @@ sleep infinity
...
+    secretName: {{ .Values.existingSecret }}
 {{- end }}
+{{- if .Values.configMap }}
+- name: ext-files
+  configMap:
+    name: {{ .Values.configMap }}
+{{- else if .Files.Glob "files/*" }}
+- name: local-files
+  configMap:
+    name: {{ print "%s-files" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/_helpers.tpl:64]:
```suggestion
    name: {{ printf "%s-files" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/configmap.yaml [hunk 2]
```diff
@@ -2,8 +2,15 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ include "common.names.fullname" . }}-files
+  name: {{ print "%s-files" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/configmap.yaml:5]:
```suggestion
  name: {{ printf "%s-files" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/scheduler/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,179 @@
+{{- if eq .Values.mode "distributed" }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: Deployment
+metadata:
+  name: {{ print "%s-scheduler" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/scheduler/deployment.yaml:5]:
```suggestion
  name: {{ printf"%s-scheduler" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/scheduler/service.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{{- if eq .Values.mode "distributed" }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ print "%s-scheduler" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/scheduler/service.yaml:5]:
```suggestion
  name: {{ printf "%s-scheduler" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/server/statefulset.yaml [hunk 5]
```diff
@@ -0,0 +1,259 @@
+{{- if eq .Values.mode "distributed" }}
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
+kind: StatefulSet
+metadata:
+  name: {{ print "%s-server" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/server/statefulset.yaml:5]:
```suggestion
  name: {{ printf"%s-server" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/server/statefulset.yaml [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/server/statefulset.yaml:6]: `namespace` is missing

File: bitnami/mxnet/templates/distributed/server/statefulset.yaml [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    app.kubernetes.io/component: server
+spec:
+  replicas: {{ .Values.server.replicaCount }}
+  podManagementPolicy: {{ .Values.server.podManagementPolicy }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: server
+  serviceName: {{ print "%s-headless" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/server/statefulset.yaml:14]:
```suggestion
  serviceName: {{ printf "%s-headless" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/server/statefulset.yaml [hunk 8]
```diff
@@ -0,0 +1,259 @@
...
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: DMLC_ROLE
+              value: "server"
+            - name: DMLC_NUM_WORKER
+              value: {{ .Values.worker.replicaCount | quote }}
+            - name: DMLC_NUM_SERVER
+              value: {{ .Values.server.replicaCount | quote }}
+            - name: DMLC_PS_ROOT_URI
+              value: {{ print "%s-scheduler" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/server/statefulset.yaml:147]:
```suggestion
              value: {{ printf "%s-scheduler" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/worker/statefulset.yaml [hunk 9]
```diff
@@ -0,0 +1,264 @@
+{{- if eq .Values.mode "distributed" }}
+apiVersion: {{ include "common.capabilities.statefulset.apiVersion" . }}
+kind: StatefulSet
+metadata:
+  name: {{ print "%s-worker" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/worker/statefulset.yaml:5]:
```suggestion
  name: {{ printf "%s-worker" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/worker/statefulset.yaml [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.worker.replicaCount }}
+  podManagementPolicy: {{ .Values.worker.podManagementPolicy }}
+  {{- if .Values.worker.updateStrategy }}
+  strategy: {{- toYaml .Values.worker.updateStrategy | nindent 4 }}
+  {{- end }}
+  serviceName: {{ print "%s-headless" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/worker/statefulset.yaml:21]:
```suggestion
  serviceName: {{ printf "%s-headless" (include "common.names.fullname" .) }}
```

File: bitnami/mxnet/templates/distributed/worker/statefulset.yaml [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: DMLC_ROLE
+              value: "worker"
+            - name: DMLC_NUM_WORKER
+              value: {{ .Values.worker.replicaCount | quote }}
+            - name: DMLC_NUM_SERVER
+              value: {{ .Values.server.replicaCount | quote }}
+            - name: DMLC_PS_ROOT_URI
+              value: {{ print "%s-scheduler" (include "common.names.fullname" .) }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/worker/statefulset.yaml:152]:
```suggestion
              value: {{ printf "%s-scheduler" (include "common.names.fullname" .) }}
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Thanks @migruiz4  Could you have a look to my comments?

File: bitnami/mxnet/values.yaml [hunk 12]
```diff
@@ -179,125 +125,24 @@ cloneFilesFromGit:
...
   ##
   enabled: false
+  ## @param persistence.storageClass discourse & sidekiq data Persistent Volume Storage Class
+  ## If defined, storageClassName: <storageClass>
+  ## If set to "-", storageClassName: "", which disables dynamic provisioning
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  storageClass: ""
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/mxnet/values.yaml:143]: It is used inside this helper (at pvc.yaml):
```
  {{- include "common.storage.class" (dict "persistence" .Values.persistence "global" .Values.global) | nindent 2 }}
```

File: bitnami/mxnet/templates/distributed/scheduler/deployment.yaml [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+          {{- if not .Values.diagnosticMode.enabled }}
+          {{- if .Values.scheduler.livenessProbe.enabled }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.scheduler.livenessProbe "enabled") "context" $) | nindent 12 }}
+            tcpSocket:
+              port: mxnet
+          {{- else if .Values.scheduler.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.scheduler.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.scheduler.readinessProbe.enabled }}
+          readinessProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.scheduler.readinessProbe "enabled") "context" $) | nindent 12 }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/distributed/scheduler/deployment.yaml:167]: Not sure if this is correct, but the scheduler entrypoint by default is `sleep infinity` and there is nothing listening in `mxnet` port so the container is never ready

Comment (fmulero) [inline bitnami/mxnet/templates/distributed/scheduler/deployment.yaml:167]: Not sure if this is correct, but the scheduler entrypoint by default is `sleep infinity` and there is nothing listening in `mxnet` port so the container is never ready

**Review by fmulero — CHANGES_REQUESTED** [review]
Please take a look into my comments

File: bitnami/mxnet/templates/standalone/deployment.yaml [hunk 14]
```diff
@@ -0,0 +1,233 @@
...
+        {{- include "common.tplvalues.render" (dict "value" .Values.standalone.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- include "mxnet.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.standalone.hostAliases }}
+      hostAliases: {{- include "common.tplvalues.render" (dict "value" .Values.standalone.hostAliases "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.standalone.updateStrategy }}
+      updateStrategy: {{- toYaml .Values.standalone.updateStrategy | nindent 4 }}
+      {{- end }}
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/mxnet/templates/standalone/deployment.yaml:37]: Replaced by `spec.strategy`

File: bitnami/mxnet/templates/distributed/scheduler/deployment.yaml (hunk 13 above)
Comment (migruiz4) [reply bitnami/mxnet/templates/distributed/scheduler/deployment.yaml:167]: I moved what the mxnet scheduler deployment already has:
https://github.com/bitnami/charts/blob/5571b65f6b54f74097792c161a47566f755cc01d/bitnami/mxnet/templates/scheduler-deployment.yaml#L95-L114

File: bitnami/mxnet/templates/standalone/deployment.yaml [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+        {{- if .Values.standalone.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.standalone.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- include "mxnet.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.standalone.hostAliases }}
+      hostAliases: {{- include "common.tplvalues.render" (dict "value" .Values.standalone.hostAliases "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.standalone.updateStrategy }}
+      strategy: {{- toYaml .Values.standalone.updateStrategy | nindent 6 }}
```
Comment (fmulero) [inline bitnami/mxnet/templates/standalone/deployment.yaml:36]: It should be in `spec.strategy` not in `spec.template.spec.strategy`

Comment (migruiz4) [reply bitnami/mxnet/templates/standalone/deployment.yaml]: Nice catch! My bad, I hadn't noticed it was under the wrong spec

**Review by fmulero — DISMISSED** [review]
LGTM

**Pull request merged**